mcnode/api.go

```diff
@@ -459,6 +460,41 @@ loop:
 	}
 }
 
+// Get /data/merge/{peerId}
+// Merges raw data objects from peerId
+func (node *Node) httpMergeData(w http.ResponseWriter, r *http.Request) {
+	vars := mux.Vars(r)
+	peerId := vars["peerId"]
+
+	pid, err := p2p_peer.IDB58Decode(peerId)
+	if err != nil {
+		apiError(w, http.StatusBadRequest, err)
+		return
+	}
+
+	keys := make(map[string]Key)
+
+	scanner := bufio.NewScanner(r.Body)
+	for scanner.Scan() {
+		err = node.mergeObjectKey(scanner.Text(), keys)
```

Comment:
hmm maybe I'm misreading this but you're iterating through the body of a GET request? This is technically allowed but semantically not quite right